src/getMongoLocker.ts

```diff
@@ -30,7 +39,19 @@ export default async (collection: Collection) => {
 
         const acquireLock = async () => {
             try {
-                await collection.insertOne({ _id: key, lockId, acquiredAt: new Date() });
+                const now = new Date();
+                /**
+                 * 3 possible situations
+                 * a) There is no lock -> upsert will create the document
+                 * b) There is an expired lock -> expiresAt: { $lte: now } will match and the document will be replaced
+                 * c) There is an unexpired lock -> expiresAt: { $lte: now } will not match,
+                 * database will try to insert the new one, but it will fail on duplicate key (_id) error
+                 */
+                await collection.replaceOne(
+                    { _id: key, expiresAt: { $lte: now } },
+                    { lockId, expiresAt: new Date(now.getTime() + expireIn) },
```

Comment:
One more comment here - `replaceOne will use the _id from filter parameter when inserting new document`